libraries/AP_UAVCAN/AP_UAVCAN_DNA_Server.cpp

```diff
@@ -547,7 +547,7 @@ void trampoline_handleNodeInfo(const uavcan::ServiceCallResult<uavcan::protocol:
     uavcan::copy(resp.getResponse().hardware_version.unique_id.begin(),
                  resp.getResponse().hardware_version.unique_id.end(),
                  unique_id);
-    snprintf(name, ARRAY_SIZE(name), "%s", resp.getResponse().name.c_str());
+    snprintf(name, ARRAY_SIZE(name), "%.14s", resp.getResponse().name.c_str());
```

Comment:
why are we snprintf'ing "%s"?  Does handleNodeInfo() needs a copy of the string? otherwise we could pass the bare pointer. If it does, just change this to strncpy().